### PR TITLE
SONAR-6812 Improve format of generated UUIDs for better usage of Lucene

### PR DESCRIPTION
**Pull request by simonbrandhof**
See https://jira.sonarsource.com/browse/SONAR-6812

### CODE REVIEW

File: sonar-core/src/main/java/org/sonar/core/util/UuidFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * used the day {@code UuidFactoryImpl} is moved outside module sonar-core.
+ * See https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java
+ */
+public class UuidFactoryImpl implements UuidFactory {
+
+  /**
+   * Should be removed as long {@link Uuids} is not used anymore. {@code UuidFactoryImpl}
+   * should be built by picocontainer through a public constructor.
+   * */
+  public static UuidFactory INSTANCE = new UuidFactoryImpl();
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/util/UuidFactoryImpl.java]: MAJOR Make this "public static INSTANCE" field final [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1444)
MAJOR Make INSTANCE a static final constant or non-public and provide accessors if needed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
MINOR Rename this field "INSTANCE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS3008)

File: sonar-core/src/main/java/org/sonar/core/util/MacAddressProvider.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    // only static stuff
+  }
+
+  private static boolean isValidAddress(@Nullable byte[] address) {
+    if (address == null || address.length != BYTE_SIZE) {
+      return false;
+    }
+    for (byte b : address) {
+      if (b != 0x00) {
+        return true; // If any of the bytes are non zero assume a good address
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/util/MacAddressProvider.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 2 major
* MINOR 4 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/45c81085aef1a9ea08bec8a614ccb8e1c4c3bb31/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L217) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForCreation(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/45c81085aef1a9ea08bec8a614ccb8e1c4c3bb31/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L103) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/45c81085aef1a9ea08bec8a614ccb8e1c4c3bb31/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForUpdate(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/45c81085aef1a9ea08bec8a614ccb8e1c4c3bb31/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L125) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/5ed414ce7d3beaaeae19c5be1824c3e2ed5aadc6/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/5ed414ce7d3beaaeae19c5be1824c3e2ed5aadc6/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L217) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForUpdate(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/5ed414ce7d3beaaeae19c5be1824c3e2ed5aadc6/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L125) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForCreation(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/5ed414ce7d3beaaeae19c5be1824c3e2ed5aadc6/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L103) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/a9fee0f26a009fae806e295f3a33e0ffce5b2881/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/a9fee0f26a009fae806e295f3a33e0ffce5b2881/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L217) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForUpdate(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/a9fee0f26a009fae806e295f3a33e0ffce5b2881/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L125) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForCreation(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/a9fee0f26a009fae806e295f3a33e0ffce5b2881/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L103) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/bb33cfa5c0e52d342e35f048f169c50d2c1e360e/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L217) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForUpdate(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/bb33cfa5c0e52d342e35f048f169c50d2c1e360e/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L125) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Method 'PermissionTemplateService.validateTemplateNameForCreation(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/bb33cfa5c0e52d342e35f048f169c50d2c1e360e/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L103) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/bb33cfa5c0e52d342e35f048f169c50d2c1e360e/server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java#L228) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Comment by teryk** [comment]
few comments then …
lgtm-pulp-fiction

**Comment by simonbrandhof** [comment]
@teryk @sns-seb I applied your feedback. Thanks.

**Pull request closed without merging**